### DISCUSSION TITLE
Local variables and loops in functions and callbacks

### DISCUSSION THREAD

**Discussion by HitecExports** [slint-ui/slint#4838] [Q&A] [answered]
Hello!

Are there any plans to add the possibility of using local variables in functions and callbacks?

Is it possible to implement loops in Slint functions and callbacks?

Is it would be possible to access functions of Conditional Elements from the parent container function?

**Comment by ogoffart** [comment, marked as answer]
In general, we're trying to keep the Slint language as not a programming language, but rather a description language.
For this reason, we don't want a turing complete language. So I'd rather not add loops.
More complex things needs to be done in native code.


> Are there any plans to add the possibility of using local variables in functions and callbacks?

This is something that makes sense: https://github.com/slint-ui/slint/issues/2752
